release/models/platform/openconfig-platform.yang

```diff
@@ -403,12 +409,51 @@ module openconfig-platform {
 
     leaf location {
       type string;
+      status deprecated;
       description
         "System-supplied description of the location of the
         component within the system. This could be a bay position,
         slot number, socket location, etc. For component types that
         have an explicit slot-id attribute, such as linecards, the
-        system should populate the more specific slot-id.";
+        system should populate the more specific slot-id.
+
+        This leaf is deprecated and replaced by parent-position.";
+    }
+
+    leaf install-position {
+      type string;
+      description
+        "System-supplied index to a position where this component is
+        installed.  The position may be referred in device documenation
+        as a port, slot, bay, socket, etc.  This string must only
+        indicate the name of the position, and not any indication of
+        the name of the parent component within the system.  Instead,
+        parent component name should be present in the 'parent' leaf.
+
+        Typically the install-position is a number, but it is observed
+        that some devices may use letters or alphanumerics.  The
+        position name should be the same name used to physically
+        identify the position in documentation or printed on the
+        device.
+
+        Any component which is removable is expected to have
+        an install-position.  The ancestor component 
+
+        For component types that have an explicit slot-id attribute,
+        such as LINECARD, the system should populate the more
+        specific slot-id.";
+    }
+
+    leaf install-component {
+      type string;
+      description
+        "A component may have zero or more intermediate component
+        ancestors which do not represent the install-position.  This
+        leaf represents the ancestor component which contains the
+        'install-position'.  For example, in the component tree
```

Comment:
Could we end this example by saying that `install-position` would be 2 and `install-component` would be "CPU1" (or whatever other line card name is used)?

We should also probably state that this leaf will hold the name of the ancestor (not the type or anything else).

Reply:
done